sdk/tests/client/signing/nft.rs

```diff
@@ -44,7 +44,10 @@ async fn nft_reference_unlocks() -> Result<()> {
                 .with_coin_type(SHIMMER_COIN_TYPE)
                 .with_range(0..1),
         )
-        .await?[0]
+        .await?
+        .into_iter()
+        .next()
+        .unwrap()
```

Comment:
You really prefer `.into_iter().next().unwrap()` to `[0].clone()`? For tests?

Reply:
I'd just add a clone(), this is so ugly

Reply:
It's the principle of the thing

Reply:
Idk, this makes it a bit unreadable IMO. [0].clone() is pretty straightforward. Happy to see what others think.

Reply:
I'm pretty opinionated about that, I'll do the change back if you want

Reply:
So am I >:C I don't care that much about making the tests efficient but you changed the function in the account too!!!

Reply:
don't even need to clone, addresses are copy

Reply:
wen review the PR and see that they are not

Reply:
another alternative would be to use `.nth(0).unwrap()`

Reply:
You'd still need to clone

Reply:
not with `.into_iter()` as it is currently. I feel like you want it more readable and have it indexed, `.nth(0).unwrap()` gives you that.

Reply:
I don't see how the longer version is better

Reply:
well, technically it is better because it doesn't have to clone :trollface:

Reply:
but since it's just for testing I don't care ... do as you please :smile:

Reply:
I think that for example we should absolutely not do that, for tests I still wouldn't as there is no reason to be efficient and it just makes it way harder to read and understand , for actual code I don't mind

Reply:
Will do